crates/bevy_ui/src/render/mod.rs

```diff
@@ -161,19 +163,100 @@ pub struct ExtractedUiNodes {
     pub uinodes: Vec<ExtractedUiNode>,
 }
 
+pub fn extract_atlas_uinodes(
+    mut extracted_uinodes: ResMut<ExtractedUiNodes>,
+    images: Extract<Res<Assets<Image>>>,
+    texture_atlases: Extract<Res<Assets<TextureAtlas>>>,
+    ui_stack: Extract<Res<UiStack>>,
+    uinode_query: Extract<
+        Query<
+            (
+                &Node,
+                &GlobalTransform,
+                &BackgroundColor,
+                &ComputedVisibility,
+                Option<&CalculatedClip>,
+                &Handle<TextureAtlas>,
+                &UiTextureAtlasSprite,
+            ),
+            Without<UiImage>,
+        >,
+    >,
+) {
+    for (stack_index, entity) in ui_stack.uinodes.iter().enumerate() {
+        if let Ok((
+            uinode,
+            transform,
+            color,
+            visibility,
+            clip,
+            texture_atlas_handle,
+            atlas_sprite,
+        )) = uinode_query.get(*entity)
+        {
+            // Skip invisible and completely transparent nodes
+            if !visibility.is_visible() || color.0.a() == 0.0 {
+                continue;
+            }
+
+            let (atlas_rect, atlas_size, image) =
+                if let Some(texture_atlas) = texture_atlases.get(texture_atlas_handle) {
+                    let mut atlas_rect = *texture_atlas
+                        .textures
+                        .get(atlas_sprite.index)
+                        .unwrap_or_else(|| {
+                            panic!(
+                                "Sprite index {:?} does not exist for texture atlas handle {:?}.",
+                                atlas_sprite.index,
+                                texture_atlas_handle.id(),
+                            )
+                        });
+                    let scale = uinode.size() / atlas_rect.size();
```

Comment:
```suggestion
                // Skip loading images
                if !images.contains(&texture_atlas.texture) {
                    continue;
                }
                let scale = uinode.size() / atlas_rect.size();
```

If we perform this check here, we'll skip the scaling calculations for unloaded images.

Reply:
Changed!